file.go

```diff
@@ -20,11 +19,11 @@ func CreateFileWith(filename, contents string) {
 		Fatalln("CreateFileWith failed:", err)
 	}
 	defer f.Close()
-	_, err = io.WriteString(f, contents)
+	n, err := f.WriteString(contents)
 	if err != nil {
 		Fatalln("CreateFileWith failed:", err)
 	}
-	log.Printf("written %d bytes to %s\n", len(contents), filename) // show absolute name
+	log.Printf("written %d bytes to %s\n", n, filename) // show absolute name
```

Comment:
WriteString() returns the number of characters that have been actually written, which can be different from what we give.